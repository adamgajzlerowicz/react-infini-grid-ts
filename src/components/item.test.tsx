import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Item from './item';

describe('NothingToShow', () => {

  const component = renderer.create(<Item id={13} />);
  const tree = component.toJSON();

  it('renders', () => {
    expect(tree).toBeTruthy();
  });
  
  it('has class name', () => {
    expect(tree.children[0].props.className).toEqual('item-inner');
    expect(tree.props.className).toEqual('item-outer');
  });

  it('shows correct number', () => {
    expect(tree.children[0].children[1]).toEqual('14');
  });

  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

});
