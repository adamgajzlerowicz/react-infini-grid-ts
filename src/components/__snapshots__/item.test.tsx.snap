// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NothingToShow matches the snapshot 1`] = `
<div
  className="item-outer"
>
  <div
    className="item-inner"
  >
    Item : 
    14
  </div>
</div>
`;
